src/main/java/games/strategy/engine/lobby/server/db/LoginMetricsController.java

```diff
@@ -0,0 +1,57 @@
+package games.strategy.engine.lobby.server.db;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.time.Clock;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import games.strategy.engine.lobby.server.login.LoginType;
+
+/**
+ * Gateway for the login metrics table.
+ */
+public final class LoginMetricsController implements LoginMetricsDao {
+  private final Clock clock;
+
+  public LoginMetricsController() {
+    this(Clock.systemUTC());
+  }
+
+  @VisibleForTesting
+  LoginMetricsController(final Clock clock) {
+    assert ZoneOffset.UTC.equals(clock.getZone());
+
+    this.clock = clock;
+  }
+
+  @Override
+  public void addSuccessfulLogin(final LoginType loginType) throws SQLException {
+    final String sql = ""
+        + "insert into login_metrics "
+        + "  (login_date, anonymous_logins, registered_logins) values (?, ?, ?) "
+        + "on conflict (login_date) do update set "
+        + "  anonymous_logins=login_metrics.anonymous_logins + excluded.anonymous_logins, "
+        + "  registered_logins=login_metrics.registered_logins + excluded.registered_logins";
+    try (Connection conn = Database.getPostgresConnection();
+        PreparedStatement ps = conn.prepareStatement(sql)) {
+      setUtcDate(ps, 1, LocalDate.now(clock));
+      ps.setInt(2, (loginType == LoginType.ANONYMOUS) ? 1 : 0);
+      ps.setInt(3, (loginType == LoginType.REGISTERED) ? 1 : 0);
+      ps.execute();
+      conn.commit();
+    }
+  }
+
+  @VisibleForTesting
+  static void setUtcDate(final PreparedStatement ps, final int parameterIndex, final LocalDate localDate)
+      throws SQLException {
+    ps.setDate(parameterIndex, Date.valueOf(localDate), Calendar.getInstance(TimeZone.getTimeZone(ZoneOffset.UTC)));
```

Comment:
@DanVanAtta said:
> I'd recommend to instead use the database now() function so that the time value is computed database side. This should simplify slightly and avoids the problem of DB clock vs Server clock.

That could make testing challenging.

However, this may be a moot issue if we expand this PR to record per-user login records where the date will actually become a timestamp that has no constraints (e.g. is not a primary key as it is now).

Reply:
Testing can be made more difficult, I agree. On the other hand to enable testing we are using the wrong clock and opening up a vulnerability with timing sync issues. It is a trade-off. In general my approach is to avoid tests that would look for specific time values and just make sure you get something set and/or relative to another time. 

It can be convincing even to set a default insertion time for these columns in part just so the java code does not have to hide for it, then the time tracking columns are almost invisible to java code unless you want to query against those columns and pick them up at your choosing. So default values is perhaps an option too.

Reply:
The other reason I chose not to have the database automatically insert the timestamp is because we have two sinks for access logs: the DB and the console.  Not using the same timestamp between all sinks could lead to confusion if we ever tried to cross-reference them.